server/game/cards/03_TWI/leaders/AsajjVentressUnparalleledAdversary.ts

```diff
@@ -0,0 +1,50 @@
+import AbilityHelper from '../../../AbilityHelper';
+import { LeaderUnitCard } from '../../../core/card/LeaderUnitCard';
+import { StateWatcherRegistrar } from '../../../core/stateWatcher/StateWatcherRegistrar';
+import { CardsPlayedThisPhaseWatcher } from '../../../stateWatchers/CardsPlayedThisPhaseWatcher';
+import { Duration } from '../../../core/Constants';
+
+export default class AsajjVentressUnparalleledAdversary extends LeaderUnitCard {
+    private cardsPlayedThisPhaseWatcher: CardsPlayedThisPhaseWatcher;
+
+    protected override getImplementationId() {
+        return {
+            id: '8929774056',
+            internalName: 'asajj-ventress#unparalleled-adversary',
+        };
+    }
+
+    protected override setupStateWatchers(registrar: StateWatcherRegistrar): void {
+        this.cardsPlayedThisPhaseWatcher = AbilityHelper.stateWatchers.cardsPlayedThisPhase(registrar, this);
+    }
+
+    protected override setupLeaderSideAbilities() {
+        this.addActionAbility({
+            title: 'Attack with a unit. If you played an event this phase, it gets +1/+0 for this attack',
+            cost: AbilityHelper.costs.exhaustSelf(),
+            initiateAttack: {
+                attackerLastingEffects: {
+                    condition: (context) => this.cardsPlayedThisPhaseWatcher.getCardsPlayed((entry) => entry.playedBy === context.attacker.controller && entry.card.isEvent()).length > 0,
```

Comment:
I would move this to a helper method that can be called in both places in this file